classes/02class/exercises/c02-network09/gpavelar/README.md

```diff
@@ -0,0 +1,47 @@
+# c02-network09
+
+## Commands Execution Output
+
+- Commands for creating security groups for below table
+
+| Security Group Name | Rule    | Type            | Port Range | Source                   | Description                                  |
+| ------------------- | ------- | --------------- | ---------- | ------------------------ | -------------------------------------------- |
+| public-sg           | inbound | Custom TCP Rule | 22         | `<your-home-ip-address>` | SSH access from home to AWS                  |
+| private-sg          | inbound | Custom TCP Rule | 22         | `<public-sg-id>`         | SSH access only from public instances on AWS |
+
+```bash
+## Create security group
+aws ec2 create-security-group --group-name public-sg --description "SSH access from home to AWS" --vpc-id vpc-074c260483405f2bb --tag-specifications 'ResourceType="security-group",Tags=[{Key="Name",Value="public-sg"}]'
+# Output
+sg-0cff452c56f306f2c
+TAGS    Name    public-sg
+
+## Create security group
+aws ec2 create-security-group --group-name private-sg --description "SSH access only from public instances on AWS" --vpc-id vpc-074c260483405f2bb --tag-specifications 'ResourceType="security-group",Tags=[{Key="Name",Value="private-sg"}]'
+# Output
+sg-0d571c8f620bcc7e0
+TAGS    Name    private-sg
+
+## Allow ingress from my IP
+aws ec2 authorize-security-group-ingress --group-id sg-0cff452c56f306f2c --protocol tcp --port 22 --cidr 147.10.74.240/32
+# Output
+none
+
+## Allow ingress from public instances on my AWS
+aws ec2 authorize-security-group-ingress --group-id sg-0d571c8f620bcc7e0 --protocol tcp --port 22 -source-group sg-0cff452c56f306f2c
+# Output
+none
+```
+
+- Any extra challenges faced?
+
+> 1. Find what is `<my-home-ip-address>`. Aaahh, my ip from my machine that I am using right now... my broaden understanding...
+
+> 2. I could not use --group-name private-sg/public-sg
+>    Example:
+>    aws ec2 authorize-security-group-ingress --group-name public-sg --protocol tcp --port 22 --cidr 147.10.74.240/32
```

Comment:
Yeah, I tried to use it as well. Then, I read on [the docs](https://docs.aws.amazon.com/cli/latest/reference/ec2/authorize-security-group-ingress.html#options) that it is for `EC2-Classic` (the original release of Amazon EC2).

> [EC2-Classic, default VPC] The name of the security group. You must specify either the security group ID or the security group name in the request.

You can find more about the EC2-Classic [here](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-classic-platform.html).